sklearn/linear_model/_base.py

```diff
@@ -684,13 +684,16 @@ def fit(self, X, y, sample_weight=None):
 
         if self.positive:
             if y.ndim < 2:
-                self.coef_, self._residues = optimize.nnls(X, y)
+                self.coef_ = optimize.nnls(X, y)[0]
             else:
                 # scipy.optimize.nnls cannot handle y with shape (M, K)
+                def _nnls_0(X, y):
```

Comment:
Can we get any issue with pickling and `joblib` since the definition is within another function?

It is always quite readable to unpack the output of the parallel:

```python
coef, _ = zip(*Parallel(n_jobs=n_jobs_)(
    delayed(optimize.nnls(X, y[:, output]) for output in range(y.shape[1])
))
self.coef_ = np.vstack(coef)
```

Reply:
> Can we get any issue with pickling and joblib since the definition is within another function?

Just in case there is an issue I updated the PR unpack `outs` directly.